### PR TITLE
chore: Run Test only when necessary

### PR DESCRIPTION
**Pull request by Lightning00Blade**
**What kind of change does this PR introduce?**

CI change

**Did you add tests for your changes?**

N/A

**If relevant, did you update the documentation?**

**Summary**

We want to run our test only when relevant parts of the code are changed.
Example change to `ng-schematics` should not trigger `puppeteer` test.

**Does this PR introduce a breaking change?**

No

**Other information**

### CODE REVIEW

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -39,11 +39,12 @@ jobs:
...
         if: github.event_name != 'pull_request'
 
-  check-docs:
+  check-changes:
     name: Check documentation
     runs-on: ubuntu-latest
     outputs:
-      needs_deploying: ${{ steps.needs_deploying.outputs.value }}
+      changes: ${{ steps.changes.outputs.changes }}
+      needs_deploying: ${{ steps.changes.outputs.website }}
```
Comment (OrKoN) [inline .github/workflows/ci.yml]: do we still need `needs_deploying`?

File: .github/workflows/ci.yml [hunk 2]
```diff
@@ -39,11 +39,12 @@ jobs:
         run: npm run commitlint
         if: github.event_name != 'pull_request'
 
-  check-docs:
+  check-changes:
     name: Check documentation
```
Comment (OrKoN) [inline .github/workflows/ci.yml]: name: Check which files changed?

File: .github/workflows/ci.yml (hunk 1 above)
Comment (Lightning00Blade) [reply .github/workflows/ci.yml]: We don't

File: .github/workflows/ci.yml [hunk 3]
```diff
@@ -58,6 +59,23 @@ jobs:
         run: npm ci --ignore-scripts
       - name: Build
         run: npm run docs
+      - name: Paths Changes Filter
```
Comment (OrKoN) [inline .github/workflows/ci.yml]: name: Detect changes?

File: .github/workflows/ci.yml [hunk 4]
```diff
@@ -58,6 +59,23 @@ jobs:
...
+              - 'packages/puppeteer-core/**'
+              - 'docker/**'
+              - 'test/**'
+              - 'test-d/**'
+              - 'tools/mochaRunner/**'
+            website:
+              - 'docs/**'
+              - 'website/**'
+            ng-schematics:
+              - 'packages/ng-schematics/**'
```
Comment (OrKoN) [inline .github/workflows/ci.yml:80]: I think we should trigger all jobs if workflows change (to make sure things still work after the workflow changes)

**Review by jrandolf-google — APPROVED** [review]

**Comment by OrKoN** [comment]
looks like there is some problem with the workflow

File: .github/workflows/ci.yml [hunk 5]
```diff
@@ -39,11 +39,11 @@ jobs:
         run: npm run commitlint
         if: github.event_name != 'pull_request'
 
-  check-docs:
-    name: Check documentation
+  check-changes:
+    name: Check which packaged changed
```
Comment (OrKoN) [inline .github/workflows/ci.yml]:
```suggestion
    name: Check which packages changed
```

**Review by OrKoN — APPROVED** [review]

**Pull request merged**